### PR TITLE
Major Style Revamp

### PR DESCRIPTION
**Pull request by adedayomoshood**
Semantic HTML
Update Viewport to avoid user zooming

default-view

loading

error

### CODE REVIEW

**Review by BolajiAyodeji — APPROVED** [review]

**Comment by shalvah** [comment]
I like the effort @adedayomoshood. We're going to need to discuss this change, though.

👍:
- the error messaging
- hero text colouring
- share button

🤔 :
- The background: I dislike plain white backgrounds. They hurt my eyes (and others like me)
- Centering the hero text  vertically looks nice, but I feel this could be problematic on mobile—there wouldn't be enough space to show the results on the first viewport (This is important so users know there's content there.)
- The loading text could be bigger. As it is, I can barely see it.

Also:
- how about mobile screenshots?
- when you propose a design change for a project, it's typically a good idea to explain why (ie give reasons why the design is better, or why the maintainer should switch to it—what value does it bring, what problems does it solve).

File: docs/index.html
Comment (ooade) [inline docs/index.html:6]: Cool one on importing style this way as the other way blocks rendering

File: docs/index.html
Comment (ooade) [inline docs/index.html:24]: Passing by really :walking_man: But Curious why you have another style declaration.

Comment (ooade) [reply docs/index.html:24]: May I know why?

**Comment by ooade** [comment]
BTW, I'll suggest the developer thing goes to the bottom, that is, the content comes first. Except you're building only for developers

File: docs/index.html
Comment (BolajiAyodeji) [reply docs/index.html:6]: Can you briefly explain the differences between the two methods and how it blocks rendering?
I'd like to know, please.

Comment (ooade) [reply docs/index.html:6]: Using `@import` will wait till the imported content is fetched. You can read more on that [here](https://stackoverflow.com/a/12380004/5040413)

Comment (ooade) [reply docs/index.html:6]: Well, in this case, it doesn't have much effect. It's just preferable to using link

Comment (BolajiAyodeji) [reply docs/index.html:6]: > Using `@import` will wait till the imported content is fetched. You can read more on that [here](https://stackoverflow.com/a/12380004/5040413)

But isn't that a good reason to use @import? Because normally you don't want to see the content until the font has loaded

Comment (BolajiAyodeji) [reply docs/index.html:6]: Thanks for the info and link :+1:

**Comment by adedayomoshood** [comment]
> * The background: I dislike plain white backgrounds. They hurt my eyes (and others like me)

I intend to update such that we have a dark mode, just like we have on Twitter. I love dark modes too. Since it's a Twitter Throwback, I though is should look "Twitter" as much as it can.


> * Centering the hero text  vertically looks nice, but I feel this could be problematic on mobile—there wouldn't be enough space to show the results on the first viewport (This is important so users know there's content there.)

Yes, I understand that centering the hero text vertically will be an issue when the results have been displayed, especially on mobile devices. I already put that into consideration and catered for that. Once the search is successful, Some information are removed from the screen to provide additional space for the result to be displayed on the screen. With that, the users doesn't need to scroll down before they can see results.

screenshot 2019-02-18 at 10 38 23
Mobile Screen

screenshot 2019-02-18 at 10 39 13
Mobile Screen with results


> * The loading text could be bigger. As it is, I can barely see it.

That's true. I'll try put a border around just like the error messaging.

> * how about mobile screenshots?

Mobile screenshots published in #15 I'll update this with mobile.

**Comment by shalvah** [comment]
> I intend to update such that we have a dark mode, just like we have on Twitter. I love dark modes too. Since it's a Twitter Throwback, I though is should look "Twitter" as much as it can.

I think you should hold off on this for now, until we've decided whether to go forward with this. Here are my issues:
- I don't think it should necessarily look like Twitter. I feel having a different look distinguishes it from an official Twitter product.
- The blue and white feels monochrome-y and "professional". Maybe it's just me, but I feel it detracts from the fun look I prefer. It makes the page feel dull to me, while the current design feels light-hearted and playful.🤷‍♀️
- Less contrast between the page and the Tweets. This would be even more pronounced if there's a dark mode.

Yes @ooade we could take the developer thing down. That should be in a separate PR, though.

**Comment by BolajiAyodeji** [comment]
@adedayomoshood I think you just make it a bit dark, something that works like the current UI with less contrast.
You can use some dark animated SVG for the background and keep the other styles, the button and the rest still looks great.
What do you think?

**Comment by shalvah** [comment]
Also, I'd say 'No" to a dark mode. This is a simple website, and I don't feel it's a good investment to spend time in creating an alternate pair of styles (and fitting an extra UI element to toggle "Dark Mode").

**Comment by BolajiAyodeji** [comment]
> Also, I'd say 'No" to a dark mode. This is a simple website, and I don't feel it's a good investment to spend time in creating an alternate pair of styles (and fitting an extra UI element to toggle "Dark Mode").

I meant just dark, no alternative style or toggling here.
The other styles in this context are the buttons

**Comment by adedayomoshood** [comment]
> * I don't think it should necessarily look like Twitter. I feel having a different look distinguishes it from an official Twitter product.
> * The blue and white feels monochrome-y and "professional". Maybe it's just me, but I feel it detracts from the fun look I prefer. It makes the page feel dull to me, while the current design feels light-hearted and playful.🤷‍♀️

> * Less contrast between the page and the Tweets. This would be even more pronounced if there's a dark mode.

Hmmmm okay, I get that. I used blue and white are the colors used in Twitter, it'll be easy for a user to know it is "twitter related". I believe if someone is able to create a non-twitter logo.

Alternatively, we can do a dark theme instead. Something like what we have below.

screenshot 2019-02-18 at 14 27 25

> Yes @ooade we could take the developer thing down. That should be in a separate PR, though.

The developer thing actually disappears once there's results.

**Comment by ooade** [comment]
Haa! @adedayomoshood Disappear ke? It should just be at the bottom, don't let it disappear.

**Comment by adedayomoshood** [comment]
> Haa! @adedayomoshood Disappear ke? It should just be at the bottom, don't let it disappear.

Lol! What I mean is once you see results, you may not necessarily need it any longer. Or maybe after the results list sha. 😉

**Comment by shalvah** [comment]
Hey @adedayomoshood I appreciate the time and effort you've put into this, but I prefer to keep the current design. Summary of my reason: the proposed design, while perfectly fine in its own right, feels too "formal", and is not in line with what I have in mind for this product. Thanks!

**Pull request closed without merging**
